### PR TITLE
HSEARCH-4984 Add test deserializing payload from older Avro versions

### PR DESCRIPTION
**Pull request by marko-bekhta**
https://hibernate.atlassian.net/browse/HSEARCH-4984

### CODE REVIEW

File: mapper/orm-outbox-polling/src/test/java/org/hibernate/search/mapper/orm/outboxpolling/avro/impl/EventPayloadSerializationUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+		Files.createDirectories( testDataDirectory );
+		try ( OutputStream out = Files.newOutputStream(
+				testDataDirectory.resolve( "payload-avro-" + Protocol.class.getPackage().getImplementationVersion() ) ) ) {
+			out.write( serialized );
+		}
+	}
+
+	@Test
+	public void canDeserializePreviousPayloads() throws IOException {
+		try ( Stream<Path> payloads = Files.list( payloadTestResourceLocation() ) ) {
```
Comment (yrodiere) [inline mapper/orm-outbox-polling/src/test/java/org/hibernate/search/mapper/orm/outboxpolling/avro/impl/EventPayloadSerializationUtilsTest.java:61]: A fine candidate for a `@MethodSource` after the JUnit 5 upgrade  :D

**Review by yrodiere — APPROVED** [review]
I think you can drop the test payloads for Avro versions < 1.10 since we've never released a version of Search that would use those versions to serialize these payloads.

Otherwise this looks good, thanks :)

**Review by yrodiere — CHANGES_REQUESTED** [review]
This broke the Windows build :D

**Comment by marko-bekhta** [comment]
🪟🪟🪟 🥲

**Review by yrodiere — APPROVED** [review]

**Pull request merged**